test/src/XcooBee/Core/Api/UsersTest.php

```diff
@@ -87,8 +87,65 @@ public function testGetConversations_UseConfig()
             'apiSecret' => 'testapisecret' 
         ]);
     }
-    
-    public function testSendUserMessage() 
+
+    public function testGetUserPublicKey()
+    {
+        $usersMock = $this->_getMock(\XcooBee\Core\Api\Users::class, [
+            '_request' => true,
+        ]);
+
+        $usersMock->expects($this->once())
+                ->method('_request')
+                ->will($this->returnCallback(function ($query, $params) {
+                            $this->assertEquals(['xid' => 'testXid'], $params);
+                        }));
+
+        $usersMock->getUserPublicKey('testXid');
+    }
+
+    public function testGetUserPublicKey_UseConfig() 
+    {
+        $usersMock = $this->_getMock(\XcooBee\Core\Api\Users::class, [
+            '_request' => true,
+        ]);
+
+        $usersMock->expects($this->once())
+                ->method('_request')
+                ->will($this->returnCallback(function ($query, $params, $config) {
+                            $this->assertEquals(['xid' => 'testXid'], $params);
+                            $this->assertEquals(['apiKey' => 'testapikey', 'apiSecret' => 'testapisecret'], $config);
+                        }));
+
+        $usersMock->getUserPublicKey('testXid', [
+            'apiKey' => 'testapikey',
+            'apiSecret' => 'testapisecret'
+        ]);
+    }
+
+    public function testGetUserPublicKey_NoPgpFound()
+    {
+        $usersMock = $this->_getMock(\XcooBee\Core\Api\Users::class, [
+            '_request' => true,
+            'getUserPublicKey' => $this->_createResponse(200, (object) ['data' => []]),
```

Comment:
you mock the method first and then call it. In this case you have to mock only `_request` to return `Response` object with empty data. And in this case check that `getUserPublicKey` returns `null`